src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs

```diff
@@ -68,5 +72,28 @@ public static void DirectoryCopy(string sourceDirPath, string destDirPath, bool
                 }
             }
         }
+
+        public static IEnumerable<string> GetProjectPathsFromSolutionFile(string solutionPath)
+        {
+            if (solutionPath.Contains(".sln") && File.Exists(solutionPath))
+            {
+                string solutionDir = Directory.GetParent(solutionPath).FullName;
+                IEnumerable<string> projectPaths = null;
+                try
+                {
+                    SolutionFile solution = SolutionFile.Parse(solutionPath);
+                    projectPaths = solution.ProjectsInOrder.Select(p => p.AbsolutePath);
+                }
+                catch (Exception ex)
+                {
+                    //Should include the logger here
+                    //LogHelper.LogError(ex, $"Error while parsing solution file {solutionPath} falling back to directory parsing.");
+                    projectPaths = Directory.EnumerateFiles(solutionDir, "*.csproj", SearchOption.AllDirectories);
+                }
```

Comment:
Should we log the error here, either as a warning or as an error? This would also resolve the github warning.

Reply:
Great idea, I did as other example did in Codelyzer and used an optional parameter to pass in the logger as it is based off a microsoft interface which should make it easier to pass in.